symbolic-base/src/ZkFold/Base/Algebra/Basic/Class.hs

```diff
@@ -38,6 +39,11 @@ class FromConstant a b where
 
 instance FromConstant a a
 
+instance {-# OVERLAPPING #-} FromConstant (Constant a f) (Constant a f)
```

Comment:
Maybe we can collect algebraic instances for `Constant` together below, how it's done with `(e -> a)` and standard numeric types?